How-to-Build-a-Flask-ReactJS-Application-and-Deploy-to-Okteto.md

```diff
@@ -0,0 +1,647 @@
+      .then((message) => {
+        console.log(message);
+        setAddTodo("");
+        getUpdate();
+      });
+  };
+
+  const getUpdate = () => {
+  // Automatically update the list as you submit the form
+    fetch("/api")
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+      })
+      .then((data) => setTodo(data));
+  };
+
+  return (
+    <>
+      <Form
+        input={addTodo}
+        onFormChange={handleFormChange}
+        onFormSubmit={handleFormSubmit}
+      />
+      <Card listOfTodos={todo} />
+    </>
+  );
+};
+
+```
+Straight foward enough!
+Now let's create the detail page, this page will display each todo.
+First create a new file  `detailView.js` in src/components folder and put the code below.
+```jsx
+import React, { useState, useEffect } from "react";
+import { Link, useParams } from "react-router-dom";
+import { useHistory } from "react-router-dom";
+
+export const Detail = () => {
+  const history = useHistory();
+  const [todo, setTodo] = useState([]);
+  const { id } = useParams();
+
+  useEffect(() => {
+    fetch(`/api/${id}`)
+      .then((response) => response.json())
+      .then((data) => setTodo(data));
+  }, [id]);
+
+      .then((response) => response.json())
+      .then((message) => {
+        history.push("/"); //redirect to homepage
+
+      })
+  
+  };
+
+  return (
+    <div>
+       <br/>
+      {todo.map((data) => (
+        <div key="id">Detail: {data.detail}</div>
+      ))}
+      <br />
+
+      <Link to="/">Go Back to Todo List</Link>
+    </div>
+  );
+};
+
+```
+
+Moving foward, let's update `App.js` so that we can test out what we have.
+```jsx
+import React from "react";
+import "./App.css";
+import { TodoPage } from "./components/TodoList";
+import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
+import { Detail } from "./components/detailView";
+// import { Form } from './components/form';
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <Router>
+          <Switch>
+            <Route exact path="/">
+              <TodoPage />
+            </Route>
+            <Route path="/:id">
+              <Detail />
+            </Route>
+          </Switch>
+        </Router>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+There are some tools I used but haven't installed yet, so what you will do is update your `package.json` file to look like below:
+```json
+{
+  "name": "frontend",
+  "version": "0.1.0",
+  "proxy": "http://api:5000",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "^11.2.7",
+    "@testing-library/user-event": "^12.8.3",
+    "react": "^16.13.1",
+    "react-dom": "^17.0.2",
+    "react-router": "5.2.0",
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "4.0.3",
+    "web-vitals": "^1.1.2"
+  },
+  ...
+```
```

Comment:
Can you list the packages you used? It's unethical to manually add packages.

Please send a command such as `npm install x y z`

Reply:
@khabdrick, can you fix this?

Reply:
My bad, I thought I already committed the change
I have fixed it now!